src/main/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessor.java

```diff
@@ -516,4 +547,71 @@ private void retryableExecuteAgent(
             )
         ));
     }
+
+    /**
+     * Retryable method to perform direct sentence highlighting inference
+     */
+    private void retryableInferenceSentenceHighlighting(
+        final SentenceHighlightingRequest inferenceRequest,
+        final int retryTime,
+        final ActionListener<List<Map<String, Object>>> listener
+    ) {
+        try {
+            MLInputDataset inputDataset = new QuestionAnsweringInputDataSet(inferenceRequest.getQuestion(), inferenceRequest.getContext());
+            MLInput mlInput = new MLInput(FunctionName.QUESTION_ANSWERING, null, inputDataset);
+
+            mlClient.predict(inferenceRequest.getModelId(), mlInput, ActionListener.wrap(mlOutput -> {
+                try {
+                    List<Map<String, Object>> result = processHighlightingOutput((ModelTensorOutput) mlOutput);
+                    listener.onResponse(result);
+                } catch (Exception e) {
+                    listener.onFailure(e);
+                }
+            }, e -> {
+                RetryUtil.handleRetryOrFailure(
+                    e,
+                    retryTime,
+                    () -> retryableInferenceSentenceHighlighting(inferenceRequest, retryTime + 1, listener),
+                    listener
+                );
+            }));
+        } catch (Exception e) {
+            listener.onFailure(e);
+        }
+    }
+
+    /**
+     * Parse the ML output for batch highlighting results
+     */
+    private List<List<Map<String, Object>>> parseBatchHighlightingOutput(MLOutput mlOutput) {
```

Comment:
I see zero error handling in this method. Can we add? 
1. when mlOutput instance of ModelTensorOutput == false
2. when highlightsList.get(0) instanceof List == false
3. when mlModelTensers.isEmpty() == true
4. when highlightsList.isEmpty() == ture
5. when docHighlithgs instanceof List == false